vhdbuilder/packer/install-dependencies.sh

```diff
@@ -195,6 +195,32 @@ EOF
 
 echo "${CONTAINER_RUNTIME} images pre-pulled:" >> ${VHD_LOGS_FILEPATH}
 
+installAndConfigureArtifactStreaming() {
+  pushd /tmp || exit $ERR_ARTIFACT_STREAMING_DOWNLOAD_INSTALL
+  # download acr-mirror proxy
+  MIRROR_PROXY_VERSION='11'
+  UBUNTU_VERSION_CLEANED="${UBUNTU_RELEASE//.}"
+  MIRROR_PROXY_URL="https://acrmirrordev.blob.core.windows.net/bin/Release-11/acr-mirror-${UBUNTU_VERSION_CLEANED}.deb"
+  
+  wget $MIRROR_PROXY_URL || exit $ERR_ARTIFACT_STREAMING_DOWNLOAD_INSTALL
+  apt_get_install 30 1 600 "./acr-mirror-${UBUNTU_VERSION_CLEANED}.deb" || exit $ERR_ARTIFACT_STREAMING_DOWNLOAD_INSTALL
+
+  echo "  - [installed] acr mirror-proxy v${MIRROR_PROXY_VERSION}" >> ${VHD_LOGS_FILEPATH}
+  rm "./acr-mirror-${UBUNTU_VERSION_CLEANED}.deb"
+  popd || exit $ERR_ARTIFACT_STREAMING_DOWNLOAD_INSTALL
+
+  sudo apt install libnl-3-dev libnl-genl-3-dev libc6 libssl3 -y
+
+  echo "  - [installed] libnl-3-dev, libnl-genl-3-dev, libc6 and libssl3" >> ${VHD_LOGS_FILEPATH}
+
+  sudo /opt/acr/tools/overlaybd/install.sh || exit $ERR_ARTIFACT_STREAMING_DOWNLOAD_INSTALL
+  sudo /opt/acr/tools/overlaybd/enable-http-auth.sh || exit $ERR_ARTIFACT_STREAMING_DOWNLOAD_INSTALL
```

Comment:
I think this is missing a step to curl the mirror to configure it's settings,

For example,

```sh
# Namespace means the registry host, _default will configure the mirror to be the default host,
# When an incoming request is received that doesn't match an accepted suffix, 
# a config will be created so that subsequent requests will not avoid calling the mirror 
# ex. docker.io request will first go to the mirror and a config will be created so that
#      subsequent requests go directly to docker.io
NAMESPACE=_default
# This controls which suffix's to enable the proxy w/
ENABLE_SUFFIX=azurecr.io
# Use 'overlaybd_vnext' to use the new artifact type
STREAM_FORMAT=overlaybd
# If true, this will edit containerd/config.toml to use overlaybd snapshotter and set cri defaults for the `--hosts-dir` setting
ENABLE_CONTAINERD=true

curl \ 
  -X PUT \
  "http://localhost:8578/config?ns=$NAMESPACE&enable_suffix=$ENABLE_SUFFIX&stream_format=$STREAM_FORMAT&enable_containerd=$ENABLE_CONTAINERD"
```

Reply:
This will go in the CSE script (separate PR), as that will actually enable it for node pools that have the streaming flag.